### DISCUSSION TITLE
weird behavior when reusing the same request block

### DISCUSSION THREAD

**Discussion by yatw** [postmanlabs/postman-flows#375] [Defect]
I have a simple work flow of 
1.  getUser()
2. If user exist (200 status code) -> assign the user admin role
3. if user not exist -> create the user , then -> assign the user admin role

If I try to reuse the same assign the user admin role block
<img width="828" alt="Screen Shot 2022-08-17 at 14 47 22" src="https://user-images.githubusercontent.com/19565177/185249180-ce1d9378-9262-4107-b251-43b6cf4a25f1.png">
It goes into an infinite loop, calling assign user admin block repeatedly

If I separate out the admin role block
<img width="905" alt="Screen Shot 2022-08-17 at 14 43 47" src="https://user-images.githubusercontent.com/19565177/185248724-d9957f83-aeeb-4750-a902-6e00bd1b1584.png">
Then it works as expected, with 2 data in the list, getUser() and assign the user admin role() is both called 2 times
<img width="1354" alt="Screen Shot 2022-08-17 at 14 46 12" src="https://user-images.githubusercontent.com/19565177/185248967-4357bae5-f401-409f-8664-3775c94e1b2f.png">

Please test and see if this is a bug.

**Comment by yatw** [comment]
Hello it has been 2 weeks, and this problem is still occurring. @saswatds I really want to make sure I brought this up to your attention. We don't need an immediate fix but we need to know this issue is not ignored.

**Reply by saswatds** [reply]
@yatw Thanks for brining this to my notice. I think I know whats going wrong here. Will try and find a fix for this.
